envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/listeners/filters/RBACFilterFactory.kt

```diff
@@ -163,23 +210,57 @@ class RBACFilterFactory(
         }
     }
 
-    private fun sourceIpPrincipals(client: String, snapshot: GlobalSnapshot): List<Principal> {
+    private fun ipFromDiscoveryPrincipals(
+        client: Client,
+        selector: String?,
+        selectorMatching: SelectorMatching?,
+        snapshot: GlobalSnapshot
+    ): List<Principal> {
         val clientEndpoints = snapshot.endpoints.resources().filterKeys { client == it }.values
```

Comment:
I think we should simply get specific key from the map, instead of iterating over all keys and filtering it. I know it is from previous PR, but still worth fixing IMO.